File: edutech-backend/models/quiz.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class Question(BaseModel):
    question_text: str
    options: List[str]
    correct_answer: int  # index of correct option
    difficulty: str  # 'easy', 'medium', 'hard'

class Quiz(BaseModel):
    id: int
    title: str
    subject: str
    questions: List[Question]
    created_at: datetime
    updated_at: datetime

class QuizAttempt(BaseModel):
    id: int
    user_id: int
    quiz_id: int
    answers: List[int]  # indices of selected answers
    score: int
    completed_at: datetime
    adaptive_level: str  # for adaptive logic

class QuizResult(BaseModel):
    quiz_id: int
    score: int
    total_questions: int
    percentage: float
    feedback: str

File: edutech-frontend/types/quiz.ts
export interface Question {
  id?: string;
  question_text: string;
  options: string[];
  correct_answer: number;
  difficulty: string;
}

export interface Quiz {
  id?: string;
  title: string;
  description: string;
  subject: string;
  questions: Question[];
  created_at?: string;
  updated_at?: string;
}

export interface QuizAttempt {
  id?: string;
  user_id: string;
  quiz_id: string;
  answers: number[];
  score?: number;
  completed_at?: string;
  adaptive_level?: string;
}

export interface QuizResult {
  quiz_id: string;
  score: number;
  total_questions: number;
  percentage: number;
  feedback: string;
}

export interface QuizCreate {
  title: string;
  description: string;
  subject: string;
  questions: Question[];
}

export interface QuizAttemptCreate {
  quiz_id: string;
  answers: number[];
}
